src/dpp/wsclient.cpp

```diff
@@ -156,7 +156,11 @@ bool websocket_client::handle_buffer(std::string &buffer)
 						}
 		
 						state = CONNECTED;
-					} else {
+					} else if (status.size() < 3) {
```

Comment:
Could this not go at the top? Meaning the if statement above could just be `if(status[1 == "101")`? with a check above that being 
```c++
if (status.size() < 3) {
    log(ll_warning, "Malformed HTTP response on websocket");
    return false;
}
````

Reply:
to be honest we should never see this. it is the least likely branch to occur not most likely, unless discord websocket endpoints start speaking complete nonsense...